nomad/server_test.go

```diff
@@ -645,3 +645,29 @@ func TestServer_ReloadSchedulers_InvalidSchedulers(t *testing.T) {
 	currentWC = s.GetSchedulerWorkerConfig()
 	require.Equal(t, origWC, currentWC)
 }
+
+func TestServer_PreventRaftDowngrade(t *testing.T) {
+	ci.Parallel(t)
+
+	dir := tmpDir(t)
+	defer os.RemoveAll(dir)
```

Comment:
```suggestion
	dir := t.TempDir()
```

https://pkg.go.dev/testing#T.TempDir

Reply:
Nice, TIL. The `tmpDir` function is used in a bunch a places, so I won't try updating the other calls in this PR, but definitely something to keep in mind.